akka/src/main/scala/auth/handlers/AuthActionsHandler.scala

```diff
@@ -56,93 +59,46 @@ class AuthActionsHandler(service: AuthService, emailPasswordServices: EmailPassw
-      }
-  }
-           */
-          complete("???")
-        } ~ (path ("requestEmailVerify") & userRequired) { s ⇒
+        } ~ (path("startPasswordRecovery") & entity(as[StartPasswordRecover])) { spr =>
+          onSuccess(passwordRecoveryService.startRecovery(spr.email)) {
+            complete(emptyAnswer)
+          }
+        } ~ (path("checkPasswordRecovery") & authTokenCommand[PasswordRecoverCommand, CheckPasswordRecoverToken]) { _ =>
+          complete(emptyAnswer)
+        } ~ (path("recoverPassword") & authTokenExpirableCommand[PasswordRecoverCommand, FinishPasswordRecover]() & userAware) { (cmd, status) =>
+          onSuccess(passwordRecoveryService.finishRecovery(cmd._1.email, cmd._2.newPass)) { pid =>
+            val auth = status.getOrElse(AuthStatus(pid, Seq.empty, None))
+            respondWithAuth(auth) {
+              complete(auth)
+            }
+          }
+        } ~ (path("verifyEmail") & authTokenCommand[EmailVerifyCommand, EmailVerifyToken]) { cmd =>
+          onSuccess(emailVerifierService.verify(cmd._1.email)) {
+            complete(emptyAnswer)
+          }
+        } ~ (path("requestEmailVerify") & userRequired) { s ⇒
           complete(
             emailVerifierService.startVerify(s.userId)
-              .map(_ ⇒ StatusCodes.NoContent → HttpEntity.empty(ContentTypes.NoContentType))
+              .map(_ ⇒ emptyAnswer)
           )
-        } ~ path ("checkEmailAvailability") {
-          /*
-          Action.async(authBodyJson[EmailCheckAvailability]) {
-    implicit request ⇒
-      profileRegistrar
-        .isEmailRegistered(request.body.email).map {
-          case true ⇒
-            JsonApiFailure(400, "email_already_registered", "Email already registered", "auth").result
-          case false ⇒
-            NoContent
-        }
-  }
-           */
-          complete("???")
-        } ~ path ("startEmailChange") {
-          /*
-          UserRequiredAction.apply(authBodyJson[StartEmailChange]) {
-    implicit request ⇒
-      emailChangeService.start(request.userId, request.body.email)
-      NoContent
-  }
-           */
-          complete("???")
-        } ~ path ("finishEmailChange") {
-          /*
-          Action.async(authTokenExpirable[FinishEmailChange, ChangeEmailCommand]()) {
-    implicit request ⇒
-      for {
-        userId ← emailChangeService.finish(request.body.userId, request.body.newEmail)
-        aStatus ← authStatus(userId)
-      } yield aStatus.ok
-  }
-
-           */
-          complete("???")
+        } ~ (path("checkEmailAvailability") & entity(as[EmailCheckAvailability])) { eca =>
+          onSuccess(service.isEmailRegistered(eca.email)) {
+            case true => complete(emptyAnswer)
+            case false => failWith(AuthError.UserAlreadyRegistered)
+          }
+        } ~ (path("startEmailChange") & userRequired & entity(as[StartEmailChange])) { (status, cmd) =>
+          emailChangeService.start(status.userId, cmd.email)
```

Comment:
maybe we should track it? to show failures to user? e.g. `onSuccess(emailChangeService.start(...))`?